Docs: update component import styles (#2082)
vtex/faststore: apps/site/pages/docs/customizing/importing-styles-to-faststore-ui-components.mdx

export interface OurStoresProps {
  title: string
  description: string
}

export default function OurStores(props: OurStoresProps) {
  return (
    <section>
      <h2>{props.title}</h2>
      <p>{props.description}</p>
      <div>
        <SelectField
          id="select-state-store"
          label="State:"
          options={{
            newYork: 'New York',
            arizona: 'Arizona',
            massachusetts: 'Massachusetts',
            hawaii: 'Hawaii',
          }}
        />
        <SelectField
          id="select-city-store"
          label="City:"
          options={{
            newYorkCity: 'New York City',
            phoenix: 'Phoenix',
            boston: 'Boston',
            honolulu: 'Honolulu',
          }}
        />
        <Button
          variant="secondary"
          icon={<Icon name="ArrowRight" />}
          iconPosition="right"
        >
          Find Store
        </Button>
      </div>
    </section>
  )
}
```

For this section we will use the following component from the Faststore UI: [Button](/components/atoms/Button), [Icon](/components/atoms/Icon) and [SelectField](/components/molecules/select-field).

> 

Reviewer: @Mariana-Caetano is missing the images for this section guide 👀

---
title: 'Importing FastStore UI component styles'
---

import { Callout, Steps } from 'nextra-theme-docs'

<header className="hero">

# Importing FastStore UI component styles

</header>

If you are using a [FastStore UI component](/components) and want to apply its styles to your
FastStore project, follow the steps below. 

There are two use cases on how to import FastStore UI component styles:

> - [Importing styles when overriding a component](/docs/customizing/components-1#importing-styles-when-overriding-a-component)

Reviewer: ```suggestion
- [Importing styles when overriding a component](/docs/customizing/importing-styles-to-faststore-ui-components#importing-styles-when-overriding-a-component)
```

> - [Importing styles when overriding a component](/docs/customizing/components-1#importing-styles-when-overriding-a-component)

Reviewer: ```suggestion
- [Importing styles when overriding a component](/docs/customizing/components-1#importing-styles-when-overriding-a-component): For this use case we will override the ProductTitle component within the
 ProductDetails section. 
```

> - [Importing styles when creating a new section](/)
> 
> In this example, we will create a new section in the CMS called `OurStore`.

Reviewer: ```suggestion
- [Importing styles when creating a new section](/docs/customizing/importing-styles-to-faststore-ui-components#importing-styles-when-creating-a-new-section): In this example, we will create a new section in the CMS called `OurStore`.
```

## Before you begin

Ensure you have a basic understanding of creating a new section in the Headless CMS. Refer to [Create a new
Section](/docs/building-sections/creating-a-new-section) guide for more
information. Additionally, we recommend using [CSS
Modules](https://nextjs.org/docs/app/building-your-application/styling/css-modules)
for styling your new component.

## Importing styles when overriding a component

For this use case we will override the ProductTitle component within the
ProductDetails section. 

### Step 1 - Create the custom component

1. Open your FastStore project and navigate to the `components` folder 
and create the `CustomProductTitle.module.scss` file. 
This file is responsible to import the FastStore UI component style.

> 2. Add the following code to the `CustomProductTitle.module.scss` file.

Reviewer: ```suggestion
2. Add the following code to the `customProductTitle.module.scss` file.
```

```scss copy

> .CustomProductTitle {

Reviewer: ```suggestion
.customProductTitle {
```

    @import "@faststore/ui/src/components/molecules/Tag/styles.scss";
}
```
This will import the [Tag]() styles to the custom component that you are creating.

3. In the `components` folder, create the `CustomProductTitle.tsx` file.
4. Add the following to the file:

```tsx copy
import React from "react";
import { ProductTitle, Tag } from "@faststore/ui";

> import styles from "./CustomProductTitle.module.scss";

Reviewer: ```suggestion
import styles from "./customProductTitle.module.scss";
```

      </div>
    </section>
  )
}
```

For this section we will use the following component from the Faststore UI: [Button](/components/atoms/Button), [Icon](/components/atoms/Icon) and [SelectField](/components/molecules/select-field).

3. Create a file named `index.tsx` inside the components folder, and add the following code:

```tsx filename="src/components/index.tsx" copy
import OurStores from './OurStores'

export default { OurStores }
```

### Step 3: Synchronize the new section with the Headless CMS

1. In the terminal, run `faststore cms-sync`.
2. After the sync, you should be able to see
   `info: CMS synced successfully...` message. 

3. Run `yarn dev` to run the store locally.
4. Go to the VTEX Admin and access **Storefront > Headless CMS**.
5. Click on the `Home` content type. (We will add it to the Home just for testing purposes)
6. In the `Sections` tab, click the `+`, search for the new `Our Stores` section, and add it to your page.
7. Click on `Save`. To see the section on the storefront, refer to the [Preview Headless CMS changes in Development mode](/tbd) guide.

### Step 4: Add the Styles

Now that we have the section in the page, let's add some styles to it:

> 1. In the `components` folder, create the `OurStores.module.scss` file. This file will import the FastStore UI components styles.

Reviewer: ```suggestion
1. In the `components` folder, create the `ourStores.module.scss` file. This file will import the FastStore UI components styles.
```

> 2. Add the following code to the `OurStores.module.scss` file:

Reviewer: ```suggestion
2. Add the following code to the `ourStores.module.scss` file:
```

> ```scss filename="src/components/OurStores.module.scss" copy

Reviewer: ```suggestion
```scss filename="src/components/ourStores.module.scss" copy
```

  title: string
  description: string
}

export default function OurStores(props: OurStoresProps) {
  return (
    <section className={styles.ourStores}>
      <h2>{props.title}</h2>
      <p>{props.description}</p>
      <div>
        <SelectField
          id="select-state-store"
          label="State:"
          options={{
            newYork: 'New York',
            arizona: 'Arizona',
            massachusetts: 'Massachusetts',
            hawaii: 'Hawaii',
          }}
        />
        <SelectField
          id="select-city-store"
          label="City:"
          options={{
            newYorkCity: 'New York City',
            phoenix: 'Phoenix',
            boston: 'Boston',
            honolulu: 'Honolulu',
          }}
        />
        <Button
          variant="secondary"
          icon={<Icon name="ArrowRight" />}
          iconPosition="right"
        >
          Find Store
        </Button>
      </div>
    </section>
  )
}
```

You should be able to see something like:

4. Let's add more styles to it. You can customize it using our [design tokens](/docs/global-tokens/overview) and [component's data attributes](/components/atoms/button#customization).

> ```scss filename="src/components/OurStores.module.scss" copy

Reviewer: ```suggestion
```scss filename="src/components/ourStores.module.scss" copy
```
